### PR TITLE
The prioritization process: documentation

### PR DESCRIPTION
**Pull request by apiraino**
Hi, I've decided to throw this early draft and start gathering thoughts and comments, there is a bit to do here.

This PR is a work in progress for [this issue](https://github.com/rust-lang/compiler-team-prioritization/issues/6) and in the current state is mostly a goofy copypasted+reformatted version of @spastorino [master document on HackMD](https://hackmd.io/WJ0G17DHTHGgv0OW9I2PxA?view).

While doing this reformat I have perceived some conceptual "holes" of things that may need clarification and this is where I ask for more eyeballs to help me refine it (or also suggest a different structure) - so, thanks!

So, let's go with my comments...

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
     - [So you want to add a new option to rustc?](./compiler/new_option.md)
     - [Major Change Proposals](./compiler/mcp.md)
     - [Membership](./compiler/membership.md)
+    - [Prioritization](./compiler/prioritization.md)
+      - [Priority Levels](#)
+      - [Prioritization Process](./compiler/prioritization-process.md)
```
Comment (apiraino) [inline src/SUMMARY.md:23]: My proposal is to place this document under the "Priorit..." chapter, under what I see the place for the soon-to-be-merged PR [describing the priority levels](https://github.com/rust-lang/rust-forge/pull/390)

File: src/compiler/prioritization-process.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
+# Prioritization Process
+
+The prioritization WG has a process that iterates every two weeks at the end of which we fill up a document with [meeting notes](https://hackmd.io/pHb6eTZ2Sjy6XZmwXZHIBA?view) of `T-compiler` working group (TODO: what is the correct cadence of these meetings?).
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:3]: The priorit... meetings are biweekly, compiler team ones are weekly, correct? how do they interwine?

File: src/compiler/prioritization-process.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+## A typical triaging workflow
+
+#### 1. Issue gets filed on the [rust-lang repository](https://github.com/rust-lang/rust)
+
+#### 2. Release team applies labels to issues
+
+One or more "area" labels are applied to identify the area of responsability. Area labels are expressed as `A-*` (ex. "A-error-handling", "A-ffi", ... full list [here](https://github.com/rust-lang/rust/labels?q=A-))
+
+One or more "team" labels are applied to assign responsability for that issue to the relevant team(s). Team labels are expressed as `T-*` (ex. "T-cargo", "T-compiler", ... full list [here](https://github.com/rust-lang/rust/labels?q=T-))
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:13]: these are just my assumptions based on what I observe. Are they correct?

File: src/compiler/prioritization-process.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+If bisection is needed, a [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) is requested as well:
+
+* label `needs-bisection`, `needs-mcve` are added to the issue
+* label `ICEBreaker-Cleanup-Crew` is added to the issue (they can help improving the quality of the report, by producing a MCVE or the bisection)
+
+In general, if the scope and impact of the issue is straightforward to understand, it is directly prioritized or sent to the relevant area, otherwise it is  "nominated" for compiler team meeting (that is, appointed for discussion during the meeting).
+
+#### 3. Release team nominates for compiler team to further process (TODO: improve this section)
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:22]: clarifcation needed

File: src/compiler/prioritization-process.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+- [Drafting the meeting agenda](#drafting-the-meeting-agenda)
+- [Day before the meeting](#day-before-the-meeting)
+- [After the meeting](#after-the-meeting)
+
+We check issues for `T-compiler` and `libs-impl` and also stable/beta nominations for `T-rustdoc`.
+
+We are in the process of automating some of our work and we execute some other things in an async way. In particular, some Zulip topics are created when certain things happen to some issue.
+
+## Async process proposal (TODO: what is this section and what to do with it?)
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:43]: This part is copied from [this comment](https://github.com/rust-lang/compiler-team-prioritization/issues/6#issuecomment-650418768) of the issue this PR refers to.

What to do with this block? Is there some content relevant to this document?

File: src/compiler/prioritization-process.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+5. Once we do all this, some things from the triagebot script could be removed
+6. We probably want the script to integrate with triagebot
+   1. we could run `@triagebot prioritization pendings` on wednesdays mornings and have that figuring out what's pending an opening issues or bumping existing ones to notify us what we need to do
+   2. we could run `@triagebot prioritization agenda` and have that dumping out the agenda. I'm not 100% sure about this we are going to be filling the agenda as we go. So maybe it's more appropriate to have stuff like `@triagebot prioritization agenda issues-of-note`, `@triagebot prioritization agenda nominations` and things like that so the script could dump parts of the agenda we need to update.
+
+## Drafting the meeting agenda
+
+The draft is currently prepared by running a CLI tool, the [triagebot prioritization script](https://github.com/rust-lang/triagebot). Run it with `cargo run --bin prioritization`. The CLI will output a sequence of steps formatted as a markdown snippet. Each step represents a section of the meeting (TODO: is it?).
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:56]: clarify

File: src/compiler/prioritization-process.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+Note: In general for issues that not labeled with `T-compiler` or `libs-impl`, we want to first check if these labels are relevant and apply them (TODO: why? what does it mean?)
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:58]: clarify

File: src/compiler/prioritization-process.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+- [libs-impl P-high](https://github.com/rust-lang/rust/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3Alibs-impl+label%3AP-high+)
+
+### 7. Final review
+
+- Check the compiler calendar to see if there's an outstanding event to announce and add them to the agenda.
+- Add Major Changes Proposals
+
+#### Toolstate
+
+TODO: add a short description?
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:156]: something useful to define this section?

File: src/compiler/prioritization-process.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+#### Performance regressions
+
+- Add Triage Logs to the agenda
+  - https://github.com/rust-lang/rustc-perf/tree/master/triage#triage-logs
+- Check [perf regressions](http://perf.rust-lang.org/index.html).
+  - Notify involved actors.
+
+## Day before the meeting
+
+TODO: improve this section
```
Comment (apiraino) [inline src/compiler/prioritization-process.md:170]: improve this section

File: src/compiler/prioritization.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
+# Prioritization
+
+Issues prioritization and [triaging](triage-meeting.html) is the process to assign a priority value to issues based on their impact on users and developers.
```
Comment (apiraino) [inline src/compiler/prioritization.md:3]: This is a placeholder text, I'm open to some suggestions to improve it

File: src/SUMMARY.md (hunk 1 above)
Comment (spastorino) [reply src/SUMMARY.md:23]: :+1:, I'd place the process first and call it just Process

**Comment by apiraino** [comment]
@spastorino here is the second draft, heavily reworked - also by you :-)

**Comment by spastorino** [comment]
We've worked through this with @apiraino and I consider this good enough to be merged. I don't have merge privileges on this repository so would leave that up for someone else to do so.

Meanwhile is people on @rust-lang/wg-prioritization wants to take a look and check if all the procedure is comprehensible that would be great, but anyway we can keep improving the documentation as we go.

**Pull request merged**